### PR TITLE
fix: report error if the altering index doesn't support mmap

### LINKED ISSUE #29909: [Bug]: alter index succeeded but can't load collection for indexes doesn't support mmap
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

the altering index doesn't check whether the index supports mmap, and will suceed to alter, but the later load would fail

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by yah01**
this also checks the param value
fix #29909

### CODE REVIEW

File: internal/datacoord/index_service.go [hunk 1]
```diff
@@ -237,6 +239,21 @@ func (s *Server) CreateIndex(ctx context.Context, req *indexpb.CreateIndexReques
...
 
+func ValidateIndexParams(index *model.Index, key, value string) error {
+	switch key {
+	case common.MmapEnabledKey:
+		indexType := getIndexType(index.IndexParams)
+		if !indexparamcheck.IsMmapSupported(indexType) {
+			return merr.WrapErrParameterInvalidMsg("index type %s does not support mmap", indexType)
+		}
+
+		if value != "true" && value != "false" {
```
Comment (czs007) [inline internal/datacoord/index_service.go:250]: how about parse value as bool ?

File: internal/datacoord/index_service.go [hunk 2]
```diff
@@ -258,6 +276,10 @@ func (s *Server) AlterIndex(ctx context.Context, req *indexpb.AlterIndexRequest)
 
 		// update the index params
 		for _, param := range req.GetParams() {
+			if err := ValidateIndexParams(index, param.GetKey(), param.GetValue()); err != nil {
```
Comment (czs007) [inline internal/datacoord/index_service.go:279]: you can call  funcutil.GetAttrByKeyFromRepeatedKV  to get `index_type` to avoid  unnecessary checks.

Comment (czs007) [inline internal/datacoord/index_service.go:279]: GetAttrByKeyFromRepeatedKV

**Comment by yah01** [comment]
rerun ut

**Comment by yah01** [comment]
rerun ut

**Comment by yah01** [comment]
rerun ut

**Comment by yah01** [comment]
rerun ut

File: internal/datacoord/index_service.go [hunk 3]
```diff
@@ -237,6 +240,44 @@ func (s *Server) CreateIndex(ctx context.Context, req *indexpb.CreateIndexReques
 	return merr.Success(), nil
 }
 
+func ValidateIndexParams(index *model.Index, key, value string) error {
+	switch key {
+	case common.MmapEnabledKey:
+		indexType := getIndexType(index.IndexParams)
+		if !indexparamcheck.IsMmapSupported(indexType) {
+			return merr.WrapErrParameterInvalidMsg("index type %s does not support mmap", indexType)
+		}
```
Comment (congqixia) [inline internal/datacoord/index_service.go:249]: make mmap enabled false for mmap not support index will also return error here. sounds a little bit tricky

Comment (yah01) [reply internal/datacoord/index_service.go:249]: this means mmap feature is disabled for the given index type, so we should forbid all operations on the feature, no matter enabling or disabling the mmap

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**